src/components/SelectionList.tsx

```diff
@@ -37,19 +38,28 @@ const SelectionList = (props: ListProperties): React.JSX.Element => {
         onPress={props.hide}
       >
         <View
+          onLayout={({ nativeEvent }) => {
+            setCurrentListWidth(nativeEvent.layout.width);
+          }}
           style={[
             style.list,
             props.styles.list,
             windowHeight > windowWidth
               ? {
                   left: props.styles.list?.alignSelf === 'center'
                     ? 0
-                    : props.listX,
-                  width: props.listWidth,
+                    : props.styles.list?.width
+                    ? props.selectorRect.x
+                      + (typeof props.selectorRect.width === 'string'
+                        ? Number(props.selectorRect.width.slice(0, -1)) * windowWidth
+                        : props.selectorRect.width
+                        - currentListWidth) / 2
+                    : props.selectorRect.x,
```

Comment:
Essentially, the previous issue was that if the dropdown list had a different width from its selector box, aka the user used a custom style to make a custom width for the dropdown list, then the dropdown list would've aligned to the left side of the selector box, whereas it's supposed to align center by default. Additionally, it needed to handle the case where the user provides a percentage instead of a number, so that needed to be handled as well via scaling it with the window width.

Reply:
Strangely, now if I try to test it, removing the part that handles the possibility of percentage doesn't change anything, even though the example app uses percentage width for some dropdowns. My guess is that it might be because of some of the past merges had added unintended changes and I haven't noticed, but I could be wrong on that since it's been a while that I've worked on this.

Reply:
Also I just realized, I think this change was supposed to be in the previous PR that was merged, but it was somehow undone at some point during all the merging.